src/components/Nav/Nav.Props.ts

```diff
@@ -78,17 +78,21 @@ export interface INavLink {
   isExpanded?: boolean;
 
   /**
-   * Any additional properties to apply to the rendered links.
+   * (Optional) Aria label for nav link
    */
-  [propertyName: string]: any;
+  ariaLabel?: string;
 
   /**
-   * (Optional) Aria label for nav link
+   * (Optional) Meta info for the link, does not involving rendering.
    */
-  ariaLabel?: string;
+  key?: string;
 
   /**
-   * (Optional) Title for nav link
+   * (Optional) title for tooltip or description
    */
   title?: string;
+  /**
+   * Any additional properties to apply to the rendered links.
+   */
+  [propertyName: string]: any;
```

Comment:
this is ambiguous and not sure where it's being used. I think if you intend to add attributes to the link, we should call it 

```
linkProps?: React.HTMLProps<HTMLAnchorElement>
```